### ISSUE TITLE
Bad gif causes access violation on bad free

### ISSUE THREAD

**Issue by strazzere** [kohler/gifsicle#105]
Similar to the others;

```
gdb$ run
Starting program: /home/tstrazzere/repo/gifsicle/src/gifsicle -O2 ./weekend-crashes/id:000013,sig:11,src:000009+001152,op:splice,rep:2
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
gifsicle:./accessviolationfree.gif:#0: read error: image corrupted, min_code_size too big
gifsicle:./accessviolationfree.gif:#0: read error: image corrupted, code out of range (19 times)
gifsicle:./accessviolationfree.gif:#0: read error: (not reporting more errors)
gifsicle:./accessviolationfree.gif:#0: read error: missing 6528 pixels of image data
gifsicle:./accessviolationfree.gif:#1: read error: image corrupted, min_code_size too big
gifsicle:./accessviolationfree.gif:#1: read error: missing 1344 pixels of image data
gifsicle:./accessviolationfree.gif:#2: read error: image corrupted, min_code_size too big
gifsicle:./accessviolationfree.gif:#2: read error: image corrupted, code out of range (19 times)
gifsicle:./accessviolationfree.gif:#2: read error: (not reporting more errors)
gifsicle:./accessviolationfree.gif:#2: read error: missing 1213 pixels of image data
gifsicle: (plus more errors; is this GIF corrupt?)
gifsicle: warning: too many colors, using local colormaps
  (You may want to try ‘--colors 256’.)
gifsicle:./accessviolationfree.gif: warning: some colors undefined by colormap

Program received signal SIGSEGV, Segmentation fault.
-----------------------------------------------------------------------------------------------------------------------[regs]
  RAX: 0x02000000004014D0  RBX: 0x000000000063DAB0  RBP: 0x00007FFFF6DB7010  RSP: 0x00007FFFFFFFE250  o d I t s z a p c 
  RDI: 0x000000000063DA60  RSI: 0x0000000000000000  RDX: 0x0000000000000000  RCX: 0x0000000000000100  RIP: 0x0000000000408ACF
  R8 : 0x000000000063DA9C  R9 : 0x0000000000000000  R10: 0x00000000006302B0  R11: 0x00000000000000D4  R12: 0x00007FFFF7505516
  R13: 0x000000000063DAB0  R14: 0x00000000006388B0  R15: 0x0000000000638C30
  CS: 0033  DS: 0000  ES: 0000  FS: 0000  GS: 0000  SS: 002B				
-----------------------------------------------------------------------------------------------------------------------[code]
=> 0x408acf <Gif_ReleaseUncompressedImage+31>:	call   rax
   0x408ad1 <Gif_ReleaseUncompressedImage+33>:	mov    QWORD PTR [rbx+0x8],0x0
   0x408ad9 <Gif_ReleaseUncompressedImage+41>:	mov    QWORD PTR [rbx+0x10],0x0
   0x408ae1 <Gif_ReleaseUncompressedImage+49>:	mov    QWORD PTR [rbx+0x48],0x0
   0x408ae9 <Gif_ReleaseUncompressedImage+57>:	pop    rbx
   0x408aea <Gif_ReleaseUncompressedImage+58>:	ret    
   0x408aeb:	nop    DWORD PTR [rax+rax*1+0x0]
   0x408af0 <Gif_ClipImage>:	mov    r9,QWORD PTR [rdi+0x8]
-----------------------------------------------------------------------------------------------------------------------------
Gif_ReleaseUncompressedImage (gfi=gfi@entry=0x63dab0) at giffunc.c:714
714	    (*gfi->free_image_data)(gfi->image_data);
gdb$ exploitable
Description: Access violation near NULL during branch instruction
Short description: BranchAvNearNull (13/22)
Hash: 76a4f8f99d605e9b77fb3cfab7313902.ab595681a77cdf09ec84944276667d89
Exploitability Classification: PROBABLY_EXPLOITABLE
Explanation: The target crashed on a branch instruction, which may indicate that the control flow is tainted. However, there is a chance it could be a NULL dereference.
Other tags: AccessViolation (21/22)
```

[accessviolationfree.zip](https://github.com/kohler/gifsicle/files/1096359/accessviolationfree.zip)

**Comment by kohler** [comment]
Hi, thanks for fuzzing gifsicle!!

However, I can't replicate this. I ran `gifsicle -O2 accessviolationfree.gif -o x.gif` under valgrind. At current HEAD (a0a365136f44e5519f7f486b00a67387f641d0e8), I get no warnings or errors at all. At previous HEAD (0e02f7b62a9a3a344c34f92ddb7e178ad3b3e3ff), I get some uninitialized-value warnings, but no crash.

What version of gifsicle are you fuzzing?

**Comment by strazzere** [comment]
This is using HEAD, compiling with gcc 4.9. Some of these don't /always/ hit if memory isn't in all the correct areas.

**Comment by kohler** [comment]
What do you mean HEAD, which commit. Can you run under valgrind. I can run this command

`./src/gifsicle accessviolationfree.gif -o /dev/null`

1000 times with no segfault (exit status 1), whether compiled under gcc-5.4.0 on an Ubuntu VM, gcc-4.8.3 on an Amazon Linux VM, or clang (Apple LLVM 8.1.0) on a mac; and with no valgrind errors.

**Comment by strazzere** [comment]
Odd, this may have been one of the issues that was fixed in a previous fuzz case. Sorry about that - I was under the impression that I tested them all at the latest commit 0e02f7b62a9a3a344c34f92ddb7e178ad3b3e3ff - I'll rebase and run the corpus through again.

**Comment by strazzere** [comment]
This too was fixed by a0a365136f44e5519f7f486b00a67387f641d0e8 - however is preproducable before it.

**Comment by kohler** [comment]
Also fixed more directly by 1638a43201436f796bdf46d33b87468c089e90b8
